src/Data/Singletons/TH/Options.hs

```diff
@@ -235,20 +258,36 @@ promoteTySym name sat
 promoteClassName :: Name -> Name
 promoteClassName = prefixName "P" "#"
 
+promoteDataTypeOrConName :: Name -> Name
+promoteDataTypeOrConName nm
+  | nm == typeRepName               = typeKindName
```

Comment:
This special case (which promoted `TypeRep` to `Type`) was inherited from the old code in `D.S.Promote.Type.promoteType_NC`, which I copied over wholesale. But to be honest, I'm really not sure why this special case exists in the first place. It's not like we use `TypeRep` as the right-hand side of a `Demote` instance anywhere. And even if we did, I'm not sure that we'd want this to special case to always be in effect. (After all, we don't have special cases for `Natural`/`Text` always be in effect.)

I'm inclined to just remove this special case. What do you think?

Reply:
`git blame` reveals that commit c1344d6 introduced this special case. Unfortunately, that commit doesn't really contain any insights as to _why_ it's special-cased.

Reply:
Then remove it. I think this was a half-hearted attempt to make more types "just promote" -- but it clearly never worked.